src/careamics/config/data_model.py

```diff
@@ -22,6 +23,30 @@
 from .transformations.xy_random_rotate90_model import XYRandomRotate90Model
 from .validators import check_axes_validity, patch_size_ge_than_8_power_of_2
 
+
+def np_float_to_str(x: float) -> str:
+    """Return a string representation of a float.
+
+    In particular, this method is used to serialize floats to strings, allowing
+    numpy.float32 to passed in the Pydantic model and written to a yaml file as str.
+
+    Parameters
+    ----------
+    x : float
+        Input value.
+
+    Returns
+    -------
+    str
+        String representation of the input value.
+    """
+    return str(x)
```

Comment:
Maybe we can use [`np.format_float_scientific`](https://numpy.org/doc/stable/reference/generated/numpy.format_float_scientific.html), this means we can choose the precision that is saved.

Reply:
Sounds good!